### ISSUE TITLE
Sketches run without error msg, but no window displayed - Win10/64  P ver 3.3.4

### ISSUE THREAD

**Issue by steve2718** [jdf/Processing.py-Bugs#187]
In python mode, sketches (appear to) run, no error messages are generated, but no window is displayed. Same sketch(es) written in java mode run fine. Uninstalled / re-installed, same.

![python_mode_no_window](https://user-images.githubusercontent.com/18587573/27012788-703ea902-4e8c-11e7-95c6-2a0da2565f92.JPG)

![java_mode_circles](https://user-images.githubusercontent.com/18587573/27012791-75db6bb6-4e8c-11e7-93e7-ac81a1a3c07f.JPG)

**Comment by jdf** [comment]
Are you quite sure the window isn't just hiding behind the editor?

**Comment by steve2718** [comment]
Yep. Also, I installed 3.3.4 on an iMac running OS X 10.12.5. Same behavior - java OK, python no window.

Here are two more screenshots. Notice, down in the taskbar, the little play button icon when running in java mode...and not in python mode.

![java_mode_window_in_taskbar](https://user-images.githubusercontent.com/18587573/27013382-60629178-4e97-11e7-90c7-63de2dd5867d.JPG)

![python_mode_running_no_window](https://user-images.githubusercontent.com/18587573/27013384-68149a1a-4e97-11e7-8068-34b0c7946d7c.JPG)

**Comment by jdf** [comment]
No clue.

Are you under some highly restrictive networking regime? Python Mode uses TCP/IP to connect to its sketch runner helper app.

**Comment by steve2718** [comment]
No. Just behind a router with AVG a/v&firewall on the Windows machine. AVG is usually pretty good about notifications of "suspicious behavior", i.e. an app trying to connect to a resource that AVG considers suspect. The iMac is just behind a router. I understand the argument about TCP/IP, but then why would/does the java mode work?

**Comment by jdf** [comment]
I don't know yet why Python Mode isn't working for you, but I do know that Java mode just launches the script runner on every involcation, whereas the Python Mode runner is a long-lived process that is told to run over a network connection.

**Comment by steve2718** [comment]
Firewall's the problem. Turned it off and python mode creates window. Where is the helper app so that I can create an exception?

**Comment by jdf** [comment]
The helper app is... Java, I guess. It's launched from withing Python Mode by running a class in `[sketchbook directory]/modes/PythonMode/mode/PythonMode.jar`.

**Comment by steve2718** [comment]
Thanks. I'll try to get an exception set up so things play nice with java. Will post findings / results.

**Comment by jdf** [comment]
Yay! Thank you.

**Comment by steve2718** [comment]
For a PC with AVG: Open AVG app / console. Menu | Settings | General | Exceptions, then Add path to PythonMode.jar

**Comment by steve2718** [comment]
Now I'll have to figure out how to do the equivalent on the Mac.

**Comment by jdf** [comment]
I'm so glad that works.

**Comment by steve2718** [comment]
Looks like I spoke too soon. I created exceptions for PythonMode.jar and javaw.exe. The firewall is still blocking TCP from javaw.exe. There's also some ICMPv6 blocking going on. What does work is to turn the firewall off (momentarily) start Processing in Python mode, then restart the firewall.

**Comment by atrivedi397** [comment]
Same problem occurs on CentOS 7 in python mode . Any clue to resolve it ?

**Comment by skilfullycurled** [comment]
I just [updated a similar issue today](https://github.com/jdf/processing.py/issues/254#issuecomment-616713198) with a solution someone found in the forums.  The issue was filed for macOS but perhaps it will be useful for other operating systems as well.
